config/reference_config.go

```diff
@@ -61,7 +61,7 @@ type ReferenceConfig struct {
 	Params         map[string]string `yaml:"params"  json:"params,omitempty" property:"params"`
 	invoker        protocol.Invoker
 	urls           []*common.URL
-	Generic        bool   `yaml:"generic"  json:"generic,omitempty" property:"generic"`
+	Generic        string `yaml:"generic"  json:"generic,omitempty" property:"generic"`
```

Comment:
should use protocol instead of expanding the enum of generic ?

Reply:
IMO, `generic` indicates how the parameters are serialized, so it can be different from protocol.